### PR TITLE
Add connection prometheus with tls configuration

### PR DESCRIPTION
**Pull request by fulviodenza**
This feature adds an example for connections pointing to a prometheus instance protected by (m)TLS.

### CODE REVIEW

File: examples/OktoResources/ResourceRelations/Connection/06.external-target-prometheus-connection-tls-config.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        (avg by (instance) 
+          (rate(node_cpu_seconds_total{job="node",mode="idle"}[1m])) * 100)
+        > 70
+      endpoint:
+        insecureValue: |
+          address: http://prometheus:9090
+          serverCertificateAuthority: | {{ .Values.prometheus.endpoint.serverCertificateAuthority | nindent 12 }}
+          clientKeyPair:
+            certificate: | {{ .Values.prometheus.endpoint.clientCertificate | nindent 14 }}
+            key: | {{ .Values.prometheus.endpoint.clientKey | nindent 14 }}
```
Comment (alvarovn) [inline examples/OktoResources/ResourceRelations/Connection/06.external-target-prometheus-connection-tls-config.yaml]: Let's make an example for a prometheus config with this. Let's not use {{ .Values... }} there, and use throwaway certs.

**Review by alvarovn — CHANGES_REQUESTED** [review]

File: examples/OktoResources/ResourceRelations/Connection/06.external-target-prometheus-connection-tls-config.yaml (hunk 1 above)
Comment (alvarovn) [reply examples/OktoResources/ResourceRelations/Connection/06.external-target-prometheus-connection-tls-config.yaml]: To clear up ambiguity: Let's make a prometheus config typed connection instead. The external target doesn't get directly affected by this.

File: examples/OktoResources/ResourceRelations/Connection/07.prometheus-config-connection.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
+apiVersion: blocks/v1beta1
+kind: Connection
+metadata:
+  name: conn-prom-config
+  namespace: default
+  labels:
+    site.nbycomp.com/{{ .Values.placement.site.label }}: "true"
+    kind: K8sCluster
```
Comment (alvarovn) [inline examples/OktoResources/ResourceRelations/Connection/07.prometheus-config-connection.yaml]: We can remove the labels

File: examples/OktoResources/ResourceRelations/Connection/07.prometheus-config-connection.yaml
Comment (alvarovn) [inline examples/OktoResources/ResourceRelations/Connection/07.prometheus-config-connection.yaml:1]: It's missing the connection type.

File: examples/OktoResources/ResourceRelations/Connection/06.external-target-prometheus-connection-tls-config.yaml
Comment (alvarovn) [inline examples/OktoResources/ResourceRelations/Connection/06.external-target-prometheus-connection-tls-config.yaml:1]: I don't think we need this connection in the examples.

Comment (fulviodenza) [reply examples/OktoResources/ResourceRelations/Connection/06.external-target-prometheus-connection-tls-config.yaml:1]: removed

File: examples/OktoResources/ResourceRelations/Connection/07.prometheus-config-connection.yaml
Comment (fulviodenza) [reply examples/OktoResources/ResourceRelations/Connection/07.prometheus-config-connection.yaml:1]: Added

File: examples/OktoResources/ResourceRelations/Connection/07.prometheus-config-connection.yaml (hunk 2 above)
Comment (fulviodenza) [reply examples/OktoResources/ResourceRelations/Connection/07.prometheus-config-connection.yaml]: removed

**Review by alvarovn — APPROVED** [review]
LGTM, le'ts squash and merge ASAP

**Pull request merged**